Manage members in app

mejo- opened the pull request:
### 📝 Summary

This PR adds functionality to manage members for a collective inside the Collectives app. This removes our dependency on the Contacts app to do so.

It's possible to add and remove members and promote/demote their membership level.

* Resolves: #308
* Resolves: #244

<!-- Write a summary of your change and some reasoning if needed -->

#### 🖼️ Screenshots

Searching for members | Changing membership level
---|---
![grafik](https://github.com/nextcloud/collectives/assets/3582805/6ebded3e-ef53-47f3-b05c-fd0c81f21ec6) | ![grafik](https://github.com/nextcloud/collectives/assets/3582805/d1b19bc0-3ff9-452d-9a2e-c41e2b9842d1)

#### 🖼️ Screencast

| 🏡 Walkthrough creating a collective and managing its members |
|---|
| [recording1.webm](https://github.com/nextcloud/collectives/assets/3582805/4ef9d177-205e-4937-98d0-c272afd35e49) |

### 🚧 TODO

- [x] Cypress tests
- [x] Don't allow to delete circle/group if current admin is member through it
- [x] Always show option to leave collective (also to admins)
- [x] Remove link to manage members in contacts
- [x] Remove dependency checks on contacts app (and close #779)
- [x] Filtering out member groups doesn't work (groups have type 16 in circle response)

### 🏁 Checklist

- [x] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [x] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [x] Tests (unit, integration and/or end-to-end) passing and the changes are covered with tests
- [x] Documentation ([README](https://github.com/nextcloud/collectives/blob/main/README.md) or [documentation](https://github.com/nextcloud/collectives/blob/main/docs/)) has been updated or is not required

mejo- commented:
@nextcloud/designers I'd appreciate feedback from you :blush: In particular on the icons used for admin/moderator/member levels and on the way the loading indicator icon is used.

marcoambrosini reviewed (approved):
From a design perspective, I'm sorry but I don't have any constructive feedback to share. It looks very neat and easy to use, congrats!! :)

juliushaertl commented on cypress/e2e/collective-members.spec.js:
```diff
@@ -0,0 +1,109 @@
+
+	describe('Manage members', function() {
+		it('Allows to add members', function() {
+			// const addMembers = ['alice', 'Bobs Group']
```
Any reason to leave this around? Shall we have a separate test for the groups?

mejo- replied:
Yes, I want to switch back to test with a group as well as soon as I resolved the following problem: for now, groups that are part of members already are not filtered out of the search results. The problem is that groups have type `16` (`TYPE_GROUP`) in circle response.

juliushaertl reviewed (commented):
Looks very nice, code also looks good, just a few commments from testing:

- [x] The search results are not keyboard accessible
- [ ] The scroll in the modal seems a bit odd with two scroll areas:
   
<img width="628" alt="Screenshot 2023-07-25 at 09 30 04" src="https://github.com/nextcloud/collectives/assets/3404133/41464403-e453-4eab-af6f-f5b359c2e072">

- [ ] Maybe we can hide the "Add users, groups  or circles ..." heading if the search is not triggered, the placeholder hint to search to add groups seems enough and the members also don't have a heading in the default state

<img width="649" alt="Screenshot 2023-07-25 at 09 32 39" src="https://github.com/nextcloud/collectives/assets/3404133/ee2eecfc-03be-45d7-9774-051605fe9b2a">

mejo- commented:
>     Maybe we can hide the "Add users, groups or circles ..." heading if the search is not triggered, the placeholder hint to search to add groups seems enough and the members also don't have a heading in the default state

@juliushaertl I would prefer to keep it. Talk behaves the same in the participants tab and I think its good to emphasize that you can search for new members to add.

max-nextcloud commented:
Looking at the screencast it seems that people, circles and groups that are already in the collective are still listed in the sections to add members. I tried in talk and it seems to filter them out. I think that would be good.

mejo- commented:
> Looking at the screencast it seems that people, circles and groups that are already in the collective are still listed in the sections to add members. I tried in talk and it seems to filter them out. I think that would be good.

This is already fixed in the PR, but I didn't update the screencast :relaxed:

nimishavijay commented:
Looks super nice! Only some minor points :)

- We generally don't use the "light" font weight very much, instead we use the regular font weight in `--color-text-maxcontrast`
- We could switch the icons for admin and moderator since the admin role has the most permissions
- Since there are 2 levels of permissions, there could be an simple explanation in a subline for all the roles to avoid confusion between admin and moderators
  - Admin: Manage collective members, view, edit, share and delete collectives
  - Moderators: View collectives, edit and share collectives if permitted
  - Members: View collectives, edit and share collectives if permitted

mejo- commented:
Thanks @nimishavijay. I already changed the font color to `--color-text-maxcontrast` and switched the icons. For the level explanations this is a bigger thing and I'd like to postpone it, as I either would have to reimplement or extend NcActionButton.
